modules/compute/gke-node-pool/main.tf

```diff
@@ -283,6 +291,22 @@ resource "google_container_node_pool" "node_pool" {
       )
       error_message = "Shared extended reservations are not supported by GKE."
     }
+    precondition {
+      condition     = contains(["SURGE"], local.upgrade_settings.strategy)
+      error_message = "Only SURGE strategy is supported"
+    }
+    precondition {
+      condition     = local.upgrade_settings.max_unavailable >= 0
+      error_message = "max_unavailable should be set to 0 or greater"
+    }
+    precondition {
+      condition     = local.upgrade_settings.max_surge >= 0
+      error_message = "max_surge should be set to 0 or greater"
+    }
+    precondition {
+      condition     = local.upgrade_settings.max_unavailable > 0 || local.upgrade_settings.max_surge > 0
+      error_message = "At least one of max_unavailable or max_surge must greater than 0"
+    }
```

Comment:
Nit: error messages end with a "." from what I see in the module and across the repo